evm_loader/lib/src/account_storage.rs

```diff
@@ -647,6 +756,13 @@ impl<'a, T: Rpc> EmulatorAccountStorage<'_, T> {
             return Ok(account);
         }
 
+        if self.is_storage_overriden(pubkey) {
+            // In case of emulated state overrides full storage we should not
+            // go to the node and return empty account here to fill it by
+            // overriden data.
+            return self.add_empty_account(pubkey);
```

Comment:
If state is overridden and code is not, here you will get an empty account with empty code?

Reply:
yes

Reply:
it works same way on geth:
- Implementation:
   - https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1006
   - https://github.com/ethereum/go-ethereum/blob/master/core/state/statedb.go#L465
- Tests where they pass code too in pair with a new state https://github.com/ethereum/go-ethereum/blob/47af69c2bc7ac28c838c5f289de23e65627c2024/eth/tracers/api_test.go#L777